webapp/javascript/components/TimelineChart/Annotations.plugin.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import * as ReactDOM from 'react-dom';
+import Color from 'color';
+import { Provider } from 'react-redux';
+import store from '@webapp/redux/store';
+import { CtxType } from './types';
+import extractRange from './extractRange';
+import AnnotationMark from './AnnotationMark';
+
+type AnnotationType = {
+  content: string;
+  timestamp: number;
+  type: 'message';
+  color: Color;
+};
+
+interface IFlotOptions extends jquery.flot.plotOptions {
+  annotations?: AnnotationType[];
+  wrapperId?: string;
+}
+
+(function ($) {
+  function init(plot: jquery.flot.plot & jquery.flot.plotOptions) {
+    plot.hooks!.draw!.push((plot, ctx: CtxType) => {
+      const options: IFlotOptions = plot.getOptions();
+
+      if (options.annotations?.length) {
+        const plotOffset: { top: number; left: number } = plot.getPlotOffset();
+        const extractedX = extractRange(plot, 'x');
+        const extractedY = extractRange(plot, 'y');
+
+        options.annotations.forEach((a: AnnotationType) => {
+          const left: number =
+            Math.floor(extractedX.axis.p2c(a.timestamp * 1000)) +
+            plotOffset.left;
+
+          renderAnnotationMark({
+            annotation: a,
+            options,
+            left,
+          });
+
+          drawAnnotationLine({
+            ctx,
+            yMin: plotOffset.top,
+            yMax:
+              Math.floor(extractedY.axis.p2c(extractedY.axis.min)) +
+              plotOffset.top,
+            left,
+            color: a.color,
+          });
```

Comment:
this is better, but from a reader pov i prefer to see a `renderAnnotationInTimeline` or something, which then can call these other 2 functions as you currently have

Reply:
>this is better, but from a reader pov i prefer to see a renderAnnotationInTimeline or something, which then can call these other 2 functions as you currently have

reworked this part, but can you please double check it to confirm that I understood you correctly

>Plus I found mark confusing. In flotjs a marking is a simple line/rectangular areas. Now you are calling mark the actual icon. That's confusing.
I renamed it to `renderAnnotationIcon`, but I'm not really happy with this decision because it's not just icon. but I can't come up anything better :) any suggestion?